src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ArrayAdapter implements CacheItemPoolInterface
```

Comment:
Expiry alone is worthless if you for instance use the cache service in a decorator for your persistence / data store ( [Example](https://doc.ez.no/display/TECHDOC/Persistence+cache) ), in such a scenario you need to invalidate cache on changes to the data.

Feel free to suggest a place to discuss this.